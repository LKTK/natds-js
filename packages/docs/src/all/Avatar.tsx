import * as React from "react";
import { Avatar } from "@naturacosmeticos/natds-web";
import { useStyles } from "./styles";

export const WrappedAvatar = () => {
  const classes = useStyles();

  return (
    <div className={classes.container}>
      <Avatar color="primary" size="huge">MS</Avatar>
    </div>
  );
};
