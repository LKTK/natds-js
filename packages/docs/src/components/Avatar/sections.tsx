import React from "react";
import { Avatar } from "@naturacosmeticos/natds-web";
import Favorite from "@material-ui/icons/Favorite";

export const colors = [
  {
    title: "Primary",
    component: <Avatar children="XD" size="standard" color="primary" />,
  },
  {
    title: "Secondary",
    component: <Avatar children="XD" size="standard" color="secondary" />,
  },
  {
    title: "Default",
    component: <Avatar children="XD" size="standard" color="default" />,
  },
];

export const types = [
  {
    title: "Image",
    component: <Avatar src={require("../../assets/images/1.png")} size="standard" color="primary" />,
  },
  {
    title: "Letter",
    component: <Avatar children="XD" size="standard" color="primary" />,
  },
  {
    title: "Icon",
    component: <Avatar children={<Favorite />} size="standard" color="primary" />,
  },
];

export const sizes = [
  {
    title: "Tiny",
    component: <Avatar src={require("../../assets/images/1.png")} size="tiny" color="primary" />,
  },
  {
    title: "Small",
    component: <Avatar src={require("../../assets/images/1.png")} size="small" color="primary" />,
  },
  {
    title: "Standard",
    component: <Avatar src={require("../../assets/images/1.png")} size="standard" color="primary" />,
  },
  {
    title: "Large",
    component: <Avatar src={require("../../assets/images/1.png")} size="large" color="primary" />,
  },
  {
    title: "Huge",
    component: <Avatar src={require("../../assets/images/1.png")} size="huge" color="primary" />,
  },
];
