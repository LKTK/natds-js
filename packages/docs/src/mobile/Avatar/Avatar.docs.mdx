import { Meta, Story, Preview } from '@storybook/addon-docs/blocks';
import { Avatar, Provider, themes } from '@naturacosmeticos/natds-rn';
import { withTheme } from '../../../addons/theme/provider';

# Avatar

---

---

## API

```jsx highlight-line="2"
import { Avatar } from '@naturacosmeticos/natds-rn';

<Avatar
  type="image"
  label=""
  icon=""
  source={require('../../assets/images/avatar.png')}
/>;
```

---

---

## Types
<Preview>
  <Provider theme={themes.natura.light}>
    <div style={{ padding: '20px' }}>
      <div style={{ width: '70px', float: 'left', marginRight: '20px' }}>
        <Avatar
          type="image"
          label=""
          icon=""
          source={require('../../assets/images/1.png')}
        />
      </div>
      <div style={{ width: '70px', float: 'left', marginRight: '20px' }}>
        <Avatar
          type="image"
          label=""
          icon=""
          source={require('../../assets/images/avatar.png')}
        />
      </div>
      <div style={{ width: '70px', float: 'left', marginRight: '20px' }}>
        <Avatar
          type="text"
          label="XD"
          icon=""
          source={{}}
        />
      </div>
      <div style={{ width: '70px', float: 'left', marginRight: '20px' }}>
        <Avatar
          type="icon"
          label=""
          icon="delete"
          source={{}}
        />
      </div>
    </div>
  </Provider>
</Preview>

## Sizes
<Preview>
  <Provider theme={themes.natura.light}>
    <div style={{ padding: '20px' }}>
      <div style={{float: 'left', marginRight: '20px' }}>
        <Avatar
          type="image"
          label=""
          icon=""
          source={require('../../assets/images/1.png')}
          avatarSizes="tiny"
        />
      </div>
      <div style={{ float: 'left', marginRight: '20px' }}>
        <Avatar
          type="image"
          label=""
          icon=""
          source={require('../../assets/images/1.png')}
          avatarSizes="small"
        />
      </div>
      <div style={{ float: 'left', marginRight: '20px' }}>
        <Avatar
          type="image"
          label=""
          icon=""
          source={require('../../assets/images/1.png')}
          avatarSizes="standard"
        />
      </div>
      <div style={{ float: 'left', marginRight: '20px' }}>
        <Avatar
          type="image"
          label=""
          icon=""
          source={require('../../assets/images/1.png')}
          avatarSizes="large"
        />
      </div>
      <div style={{ float: 'left', marginRight: '20px' }}>
        <Avatar
          type="image"
          label=""
          icon=""
          source={require('../../assets/images/1.png')}
          avatarSizes="xlarge"
        />
      </div>
    </div>
  </Provider>
</Preview>

| Name                      | Type                                            | Default                       | Description
| :--                       | :-:                                             | :-:                           | :-:                       |
| type *                    | 'image', 'text', 'icon'                         |                               | Type of Avatar (image, text or icon).
| label **                  | string                                          |                               | Initials to show as the text in the `Avatar`. Use only in Text Type.
| icon **                   | IconSource                                      |                               | Icon to display for the `Avatar`. Use only in Icon Type.
| avatarSizes               | 'tiny', 'small', 'standard', 'large', 'xlarge'  | 'standard'                    | The possible sizes of the Avatar *
| style                     | object                                          |                               | React Native style object.
| theme                     | IThemeMobile                                    |                               | Overridable theme provided from the Provider component
| color                     | string                                          |                               | Custom color for the text.
| labelStyle                | object                                          |                               | Style for the title. Use only in Text Type.
