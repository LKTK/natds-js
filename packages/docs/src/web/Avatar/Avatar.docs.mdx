import { Meta, Story, Preview } from '@storybook/addon-docs/blocks';
import { Avatar, Provider, themes } from '@naturacosmeticos/natds-web';
import { withTheme } from '../../../addons/theme/provider';
import { Delete } from '@material-ui/icons';

# Avatar

---

---

## API

```js highlight-line="2"
import { Avatar } from '@naturacosmeticos/natds-web';

<Avatar
  src={require('../../assets/images/avatar.jpg')}
/>;
```

---

---

## Types
<Preview>
  <Provider theme={themes.natura.light}>
    <div style={{ padding: '20px', display: 'flex' }}>
      <div style={{ width: '70px', marginRight: '20px' }}>
        <Avatar src={require('../../assets/images/1.jpg')} />
      </div>
      <div style={{ width: '70px', marginRight: '20px' }}>
        <Avatar src={require('../../assets/images/avatar.jpg')} />
      </div>
      <div style={{ width: '70px', marginRight: '20px' }}>
        <Avatar children="XD" />
      </div>
      <div style={{ width: '70px', marginRight: '20px' }}>
        <Avatar children={<Delete />} />
      </div>
    </div>
  </Provider>
</Preview>

---

---

## Variants
<Preview>
  <Provider theme={themes.natura.light}>
    <div style={{ padding: '20px', display: 'flex' }}>
      <div style={{ width: '70px', marginRight: '20px' }}>
        <Avatar src={require('../../assets/images/1.jpg')} />
      </div>
      <div style={{ width: '70px', marginRight: '20px' }}>
        <Avatar src={require('../../assets/images/1.jpg')} variant="rounded" />
      </div>
      <div style={{ width: '70px', marginRight: '20px' }}>
        <Avatar src={require('../../assets/images/1.jpg')} variant="square" />
      </div>
    </div>
  </Provider>
</Preview>

---

---

| Name                      | Type                                            | Default                       | Description
| :--                       | :-:                                             | :-:                           | :-:                       |
| alt                       | string                                          |                               | Used in combination with src or srcSet to provide an alt attribute for the rendered img element.
| children                  | node                                            |                               | Used to render icon or text elements inside the Avatar if src is not set.This can be an element, or just a string.
| classes                   | object                                          |                               | Override or extend the styles applied to the component.See CSS API below for more details.
| component                 | elementType                                     | 'div'                         | The component used for the root node.Either a string to use a DOM element or a component.
| imgProps                  | object                                          |                               | Attributes applied to the img element if the component is used to display an image.
| sizes                     | string                                          |                               | The sizes attribute for the img element.
| src                       | string                                          |                               | The src attribute for the img element.
| srcSet                    | string                                          |                               | The srcSet attribute for the img element.
| srcSet                    | string                                          | 'rounded', 'square', 'circle' | The shape of the avatar.
