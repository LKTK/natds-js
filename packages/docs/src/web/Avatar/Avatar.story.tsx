import * as React from 'react';
import { Avatar } from '@naturacosmeticos/natds-web';
import { number, select, text } from '@storybook/addon-knobs';
import withJest from '@decorators/jest/jest';
import withContainer from '@decorators/container/container';
import { Delete } from '@material-ui/icons';

import AvatarDocs from './Avatar.docs.mdx';

export default {
  title: 'Web|Avatar',
  component: Avatar,
  decorators: [withJest(), withContainer],
  parameters: {
    jestImportPath: 'web',
    jest: ['Avatar'],
    theme: 'web',
    docs: {
      page: AvatarDocs,
    },
  },
};

const variant: any = {
  circle: 'circle',
  rounded: 'rounded',
  square: 'square',
};

const source: any = {
  image: require('../../assets/images/1.jpg'),
  anonymous: require('../../assets/images/avatar.jpg'),
};

export const Interactive = () => (
  <Avatar
    alt={text('alt', '')}
    src={select('source', source, source.image)}
    variant={select('variant', variant, variant.circle)}
  />
);

export const Image = () => (
  <Avatar src={source.image} />
);

export const Anonymous = () => (
  <Avatar src={source.anonymous} />
);

export const Text = () => (
  <Avatar children='AB' />
);

export const Icon = () => (
  <Avatar children={<Delete />} />
);
