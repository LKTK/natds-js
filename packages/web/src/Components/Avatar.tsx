import React, { FunctionComponent, forwardRef } from "react";
import MaterialAvatar, { AvatarProps as MaterialAvatarProps } from "@material-ui/core/Avatar";
import { withTheme } from "@material-ui/styles";
import { IAvatarSizes } from "@naturacosmeticos/natds-styles";
import { IThemeWeb } from "Themes";

export type AvatarSizes = keyof IAvatarSizes;
export type AvatarColors = "primary" | "secondary" | "default" | string;

export interface IAvatarProps extends MaterialAvatarProps {

  /**
   * @optional
   * @default 'standard'
   * Size of the Avatar
   */
  size?: AvatarSizes;

  /**
   * @optional
   * @default 'primary'
   * Color of the Avatar
   */
  color?: AvatarColors;
  theme: IThemeWeb | unknown;
}

const withSizes = (size: AvatarSizes | undefined,
  // eslint-disable-next-line @typescript-eslint/no-explicit-any
  theme: any = {}) => {

  if (!size) {
    return {};
  }

  const { avatarSizes }: { avatarSizes: IAvatarSizes } = theme;

  if (!avatarSizes) {
    return {};
  }

  const value = avatarSizes[size],
    sizeInPx = `${value.size}px`;

  return {
    fontSize: value.fontSize,
    height: sizeInPx,
    width: sizeInPx,
  };
};

const withColors = (color: AvatarColors | undefined,
  // eslint-disable-next-line @typescript-eslint/no-explicit-any
  theme: any = {}) => {

  if (!color) {
    return {};
  }

  // eslint-disable-next-line @typescript-eslint/no-explicit-any
  const { palette = {} }: { palette: any } = theme,

    selectedColor = palette[color];

  if (!selectedColor || color === "default") {
    return {};
  }

  return {
    background: selectedColor.main,
    color: selectedColor.contrastText,
  };
};

export const Avatar: FunctionComponent<IAvatarProps> = forwardRef((
  props: IAvatarProps,
  // eslint-disable-next-line @typescript-eslint/no-explicit-any
  ref: any,
) => {
  const style = {
    ...withSizes(props.size, props.theme),
    ...withColors(props.color, props.theme),
    ...props.style,
  };

  return <MaterialAvatar {...props} style={style} ref={ref} />;
});

Avatar.displayName = "Avatar";

export default withTheme(Avatar);
