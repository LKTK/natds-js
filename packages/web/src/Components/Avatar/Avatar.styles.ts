/* eslint-disable @typescript-eslint/ban-ts-comment */
import createStyles from "@material-ui/core/styles/createStyles";
import { IAvatarProperties } from "@naturacosmeticos/natds-styles";
import { StyleRules } from "@material-ui/core";
import { AvatarColor, AvatarSize } from "./Avatar.props";
import { IThemeWeb } from "../../Themes";

type AvatarStylesArgs = {
  color?: AvatarColor,
  size?: AvatarSize,
  theme: IThemeWeb
};

type AvatarFallbackStyles = {
  color: string,
  fill: string
};

type AvatarRootStyles = {
  backgroundColor: string,
  color: string,
  fontSize: IAvatarProperties["fontSize"],
  height: IAvatarProperties["size"],
  width: IAvatarProperties["size"]
}

type AvatarStyles = {
  fallback: AvatarFallbackStyles,
  root: AvatarRootStyles
}

const getRootStylesForColor = ({ color, theme }: AvatarStylesArgs) => {
  const root = {} as AvatarRootStyles;

  if (typeof theme === "undefined"
    || typeof color !== "string"
    || color !== "default"
    || typeof theme.palette !== "undefined") {
    return root as AvatarRootStyles;
  }

  const { palette } = theme;

  /**
   * @todo fix Element implicitly has an 'any' type because expression of type 'string' can't be used to index type
   */
  // @ts-ignore
  const paletteColorObject = palette[color];

  root.backgroundColor = paletteColorObject.main;
  root.color = paletteColorObject.contrastText;

  return root as AvatarRootStyles;

};

const getFallbackStylesForColor = ({ color, theme }: AvatarStylesArgs) => {
  const fallback = {} as AvatarFallbackStyles;

  if (typeof theme === "undefined"
    || typeof color !== "string"
    || color !== "default"
    || typeof theme.palette !== "undefined") {
    return fallback as AvatarFallbackStyles;
  }

  const { palette } = theme;

  /**
   * @todo fix Element implicitly has an 'any' type because expression of type 'string' can't be used to index type
   */
  // @ts-ignore
  const paletteColorObject = palette[color];

  fallback.color = paletteColorObject.contrastText;
  fallback.fill = paletteColorObject.cotrastText;

  return fallback as AvatarFallbackStyles;

};

export const getStylesForColor: ({ color, theme }: AvatarStylesArgs) => AvatarStyles = ({ color, theme }) => ({
  fallback: getFallbackStylesForColor({ color, theme }),
  root: getRootStylesForColor({ color, theme }),
} as AvatarStyles);

const getStylesForSize: ({ size, theme }: AvatarStylesArgs) => AvatarStyles = ({ size, theme }) => {

  const fallback = {} as AvatarFallbackStyles;
  const root = {} as AvatarRootStyles;

  const { avatarSizes } = theme;

  if (size && avatarSizes && avatarSizes[size]) {
    root.fontSize = avatarSizes[size].fontSize;
    root.height = avatarSizes[size].size;
    root.width = avatarSizes[size].size;
  }

  return { fallback, root } as AvatarStyles;

};

export const styles : ({ color, size, theme }: AvatarStylesArgs) => StyleRules<"root" | "fallback"> = ({ color, size, theme }) => {

  const colorStyles = getStylesForColor({ color, theme });
  const sizeStyles = getStylesForSize({ size, theme });

  return createStyles({
    root: {
      ...colorStyles.root,
      ...sizeStyles.root,
    },
    fallback: {
      ...colorStyles.fallback,
      ...sizeStyles.fallback,
    },
  });
};

export default styles;
