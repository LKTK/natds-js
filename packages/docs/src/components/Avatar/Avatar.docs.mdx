import { Avatar, Provider, themes } from '@naturacosmeticos/natds-web';
import ComponentWithLabel from '../../../addons/ComponentWithLabel';
import { types, sizes } from './sections';

# Avatar

---

---

## Types

<Provider theme={themes.natura.light}>
  <ComponentWithLabel componentList={types} />
</Provider>

---

---

## Sizes

<Provider theme={themes.natura.light}>
  <ComponentWithLabel componentList={sizes} />
</Provider>

## API

```jsx highlight-line="2"
import { Avatar } from '@naturacosmeticos/natds-web';

// Image avatar
<Avatar src={require('../../assets/images/avatar.png')} size="huge" />;

// Text avatar
<Avatar childred="Jon Doe" size="huge" color="secondary" />;
```

For more advanced usages, check [Material UI Avatar docs](https://material-ui.com/components/avatars/)

---

---

## Avatar Props

| Name      |    Type     |                   Default                    |                                                    Description                                                     |
| :-------- | :---------: | :------------------------------------------: | :----------------------------------------------------------------------------------------------------------------: |
| alt       |   string    |                                              |          Used in combination with src or srcSet to provide an alt attribute for the rendered img element.          |
| children  |    node     |                                              | Used to render icon or text elements inside the Avatar if src is not set.This can be an element, or just a string. |
| classes   |   object    |                                              |             Override or extend the styles applied to the component.See CSS API below for more details.             |
| color \*  |   string    |      'primary', 'secondary', 'default'       |             The color of the Avatar, using the tokens from the currrent theme, or a custom css color.              |
| component | elementType |                    'div'                     |             The component used for the root node. Either a string to use a DOM element or a component.             |
| imgProps  |   object    |                                              |                Attributes applied to the img element if the component is used to display an image.                 |
| size \*   |   string    | 'tiny', 'small', 'standard', 'large', 'huge' |                                          The possible sizes of the Avatar                                          |
| sizes     |   string    |                                              |                                      The sizes attribute for the img element.                                      |
| src       |   string    |                                              |                                       The src attribute for the img element.                                       |
| srcSet    |   string    |                                              |                                     The srcSet attribute for the img element.                                      |
| variant   |   string    |        'rounded', 'square', 'circle'         |                                              The shape of the avatar.                                              |

- not a default Material-UI Avatar property
