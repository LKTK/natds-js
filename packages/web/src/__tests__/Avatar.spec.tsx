import React from 'react';
import { expect } from 'chai';
import renderer from 'react-test-renderer';
import { shallow } from 'enzyme';

import Avatar from '../Components/Avatar';
import { themes } from '@naturacosmeticos/natds-styles';

const label = 'A';

const MockedIcon = () => {
  return (
    <i />
  );
};

describe('Avatar component', () => {
  describe('rendering variants', () => {
    test('should match snapshot - Avatar (variant circle)', () => {
      const component = renderer.create(<Avatar variant="circle">{label}</Avatar>).toJSON();

      expect(component).to.matchSnapshot('Avatar circle snapshot');
    });

    test('should match snapshot - Avatar (variant rounded)', () => {
      const component = renderer.create(<Avatar variant="rounded">{label}</Avatar>).toJSON();

      expect(component).to.matchSnapshot('Avatar rounded snapshot');
    });

    test('should match snapshot - Avatar (variant square)', () => {
      const component = renderer.create(<Avatar variant="square">{label}</Avatar>).toJSON();

      expect(component).to.matchSnapshot('Avatar square snapshot');
    });
  });

  describe('rendering avatars types', () => {
    test('should match snapshot - Avatar with initials', () => {
      const component = renderer.create(<Avatar>AB</Avatar>).toJSON();

      expect(component).to.matchSnapshot('Avatar with initials snapshot');
    });

    test('should match snapshot - Avatar with icon', () => {
      const component = renderer.create(<Avatar><MockedIcon /></Avatar>).toJSON();

      expect(component).to.matchSnapshot('Avatar with icon snapshot');
    });

    test('should match snapshot - Avatar with image', () => {
      const image = 'source/image.png';
      const component = renderer.create(<Avatar src={image} />).toJSON();

      expect(component).to.matchSnapshot('Avatar with initials snapshot');
    });
  });

  describe('rendering avatars sizes', () => {
    test('should match snapshot - Avatar tiny', () => {
      const component = renderer.create(<Avatar size="tiny" />).toJSON();

      expect(component).to.matchSnapshot('Avatar tiny snapshot');
    });

    test('should match snapshot - Avatar small', () => {
      const component = renderer.create(<Avatar size="small" />).toJSON();

      expect(component).to.matchSnapshot('Avatar small snapshot');
    });

    test('should match snapshot - Avatar standard', () => {
      const component = renderer.create(<Avatar size="standard" />).toJSON();

      expect(component).to.matchSnapshot('Avatar standard snapshot');
    });

    test('should match snapshot - Avatar large', () => {
      const component = renderer.create(<Avatar size="large" />).toJSON();

      expect(component).to.matchSnapshot('Avatar large snapshot');
    });

    test('should match snapshot - Avatar xlarge', () => {
      const component = renderer.create(<Avatar size="xlarge" />).toJSON();

      expect(component).to.matchSnapshot('Avatar xlarge snapshot');
    });

    test('should not apply width and height if without theme', () => {
      const theme = {};
      const component = shallow(<Avatar theme={theme} />);

      expect(component.dive().prop('style')).to.equal(undefined);
    });

    test('should apply standard width and height if with theme', () => {
      const theme = themes.natura.light;
      const component = shallow(<Avatar theme={theme} />);

      expect(component.dive().prop('style')).to.deep.equal({
        width: `${theme.avatarSizes.standard}px`,
        height: `${theme.avatarSizes.standard}px`
      });
    });

    test('should apply chosen width and height if with theme', () => {
      const theme = themes.natura.light;
      const size = 'large';
      const component = shallow(<Avatar theme={theme} size={size} />);

      expect(component.dive().prop('style')).to.deep.equal({
        width: `${theme.avatarSizes[size]}px`,
        height: `${theme.avatarSizes[size]}px`
      });
    });
  });
});
