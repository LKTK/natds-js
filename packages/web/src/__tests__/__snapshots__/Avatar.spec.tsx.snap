// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Avatar component rendering avatars should match snapshot - Avatar with icon: Avatar with icon snapshot 1`] = `
<div
  className="MuiAvatar-root MuiAvatar-circle MuiAvatar-colorDefault"
>
  <i />
</div>
`;

exports[`Avatar component rendering avatars should match snapshot - Avatar with image: Avatar with initials snapshot 1`] = `
<div
  className="MuiAvatar-root MuiAvatar-circle"
>
  <img
    className="MuiAvatar-img"
    src="source/image.png"
  />
</div>
`;

exports[`Avatar component rendering avatars should match snapshot - Avatar with initials: Avatar with initials snapshot 1`] = `
<div
  className="MuiAvatar-root MuiAvatar-circle MuiAvatar-colorDefault"
>
  AB
</div>
`;

exports[`Avatar component rendering variants should match snapshot - Avatar (variant circle): Avatar circle snapshot 1`] = `
<div
  className="MuiAvatar-root MuiAvatar-circle MuiAvatar-colorDefault"
>
  A
</div>
`;

exports[`Avatar component rendering variants should match snapshot - Avatar (variant rounded): Avatar rounded snapshot 1`] = `
<div
  className="MuiAvatar-root MuiAvatar-rounded MuiAvatar-colorDefault"
>
  A
</div>
`;

exports[`Avatar component rendering variants should match snapshot - Avatar (variant square): Avatar square snapshot 1`] = `
<div
  className="MuiAvatar-root MuiAvatar-square MuiAvatar-colorDefault"
>
  A
</div>
`;
